server/src/test/java/org/elasticsearch/action/CrossProjectResolverUtilsTests.java

```diff
@@ -0,0 +1,311 @@
+            containsInAnyOrder("P1:metrics*", "P2:metrics*", "Q1:metrics*", "Q2:metrics*", "metrics*")
+        );
+    }
+
+    public void testEmptyExpressionShouldMatchAll() {
+        AuthorizedProjectsSupplier.AuthorizedProjects authorizedProjects = new AuthorizedProjectsSupplier.AuthorizedProjects(
+            "_origin",
+            List.of("P1", "P2")
+        );
+        CrossProjectReplaceableTest crossProjectRequest = new CrossProjectReplaceableTest(new String[] {}, IndicesOptions.DEFAULT);
+
+        CrossProjectResolverUtils.maybeRewriteCrossProjectResolvableRequest(remoteClusterAware, authorizedProjects, crossProjectRequest);
+
+        assertThat(crossProjectRequest.getCanonicalExpressions().keySet(), containsInAnyOrder("*"));
+        assertThat(crossProjectRequest.getCanonicalExpressions().get("*"), containsInAnyOrder("P1:*", "P2:*", "*"));
+    }
+
+    public void testWildcardExpressionShouldMatchAll() {
+        AuthorizedProjectsSupplier.AuthorizedProjects authorizedProjects = new AuthorizedProjectsSupplier.AuthorizedProjects(
+            "_origin",
+            List.of("P1", "P2")
+        );
+        CrossProjectReplaceableTest crossProjectRequest = new CrossProjectReplaceableTest(new String[] { "*" }, IndicesOptions.DEFAULT);
+
+        CrossProjectResolverUtils.maybeRewriteCrossProjectResolvableRequest(remoteClusterAware, authorizedProjects, crossProjectRequest);
+
+        assertThat(crossProjectRequest.getCanonicalExpressions().keySet(), containsInAnyOrder("*"));
+        assertThat(crossProjectRequest.getCanonicalExpressions().get("*"), containsInAnyOrder("P1:*", "P2:*", "*"));
+    }
+
+    public void test_ALLExpressionShouldMatchAll() {
+        AuthorizedProjectsSupplier.AuthorizedProjects authorizedProjects = new AuthorizedProjectsSupplier.AuthorizedProjects(
+            "_origin",
+            List.of("P1", "P2")
+        );
+        String all = randomBoolean() ? "_ALL" : "_all";
+        CrossProjectReplaceableTest crossProjectRequest = new CrossProjectReplaceableTest(new String[] { all }, IndicesOptions.DEFAULT);
+
+        CrossProjectResolverUtils.maybeRewriteCrossProjectResolvableRequest(remoteClusterAware, authorizedProjects, crossProjectRequest);
+
+        assertThat(crossProjectRequest.getCanonicalExpressions().keySet(), containsInAnyOrder("*"));
+        assertThat(crossProjectRequest.getCanonicalExpressions().get("*"), containsInAnyOrder("P1:*", "P2:*", "*"));
+    }
+
+    private static class RemoteClusterAwareTest extends RemoteClusterAware {
+        RemoteClusterAwareTest() {
+            super(Settings.EMPTY);
+        }
+
+        @Override
+        protected void updateRemoteCluster(String clusterAlias, Settings settings) {
+
+        }
+
+        @Override
+        public Map<String, List<String>> groupClusterIndices(Set<String> remoteClusterNames, String[] requestIndices) {
+            return super.groupClusterIndices(remoteClusterNames, requestIndices);
+        }
```

Comment:
Is it intentionally overwritten? Also do we need to change visibility of `RemoteClusterAware#groupClusterIndices`?